src/partition/ReceivedPartition.cpp

```diff
@@ -462,6 +485,22 @@ void ReceivedPartition::prepareBoundingBox()
     _bb.scaleBy(_safetyFactor);
     _boundingBoxPrepared = true;
   }
+
+  // Expand by user-defined bounding box in case a direct access is desired
+  if (_allowDirectAccess) {
+    auto &other_bb = _mesh->getBoundingBox();
+    _bb.expandBy(other_bb);
+    // TODO: How to treat the scale by safety factor here?
+    // The default value of 0.5 is for the current mappings
+    // still too large in comparison to what we want to do here.
+    // On the other hand, ignoring it makes the xml configuration
+    // inconsistent and does not provide any configurable option for
+    // 'bad' matching geometries.
+    //    _bb.scaleBy(_safetyFactor);
+    PRECICE_WARN("Ignoring the safety factor for bounding box initialization");
```

Comment:
What do you do then if you have bad matching geometries and a safety factor is actually desired. It would still be inconsistent with the precice-config, right? The scaling is applied anyway for all mappings a few lines above.

Reply:
Ah, OK. I did not see the scaling above. Then, here, I would not scale at all.
In my opinion, this should be the job of the user / the adapter. You define the bounding box, such that you have everything that interests you (i.e. you expand the bounding box by your own ghost layer or similar, everything else goes to another rank). Independent of how the other mesh looks like.
I agree though that we need a better solution for the preCICE config in the long run, like #1040.

Reply:
Ok,  should we then raise a warning here or should we silently skip the scaling?

Reply:
I would silently skip the scaling. Maybe we can make this explicit in the documentation of `safety-factor`?